src/web/tools/Vector.ts

```diff
@@ -0,0 +1,58 @@
+import { Directions } from '../../Directions';
+import PointerTracker from './PointerTracker';
+
+export default class Vector {
+  x: number = 0;
+  y: number = 0;
+  unitX: number = 0;
+  unitY: number = 0;
+
+  fromDirection(direction: Directions) {
+    [this.x, this.y] = [this.unitX, this.unitY] = DirectionTypeMappings.get(
+      direction
+    ) ?? [0, 0];
+
+    return this;
+  }
+
+  fromVelocity(tracker: PointerTracker, pointerId: number) {
+    this.x = tracker.getVelocityX(pointerId);
+    this.y = tracker.getVelocityY(pointerId);
+
+    const magnitude = Math.hypot(this.x, this.y);
+    if (magnitude < 1) {
```

Comment:
Even though `1` looks less confusing than `0.001`, I'd still like to give it a name (cc @j-piasecki).

We could also do something like

```js
    this.unitX = magnitude < 1 ? 0 : this.x / magnitude
    this.unitY = magnitude < 1 ? 0 : this.y / magnitude
```

But I don't have strong preferences here so it's up to you to decide which one you like more 😅